tests/pipeline/test_engine.py

```diff
@@ -1242,6 +1249,177 @@ def test_dollar_volume(self):
         expected_5 = rolling_mean((self.raw_data ** 2) * 2, window=5)[5:]
         assert_frame_equal(results['dv5'].unstack(), expected_5)
 
+    @parameter_space(returns_length=[2, 3], correlation_length=[3, 4])
+    def test_correlation_factors(self, returns_length, correlation_length):
+        """
+        Tests for the built-in factors `RollingPearsonOfReturns` and
+        `RollingSpearmanOfReturns`.
+        """
+        my_asset_column = 0
+        start_date_index = 6
+        end_date_index = 10
+
+        assets = self.asset_finder.retrieve_all(self.sids)
+        my_asset = assets[my_asset_column]
+        my_asset_filter = (AssetID() != (my_asset_column + 1))
+        num_assets = len(self.sids)
+        num_days = end_date_index - start_date_index + 1
+
+        # Our correlation factors require that their baseline asset is not
+        # filtered out, so make sure that masking out our baseline asset does
+        # not take effect. That is, a filter which filters out only our
+        # baseline asset should produce the same result as if no mask was
+        # passed at all.
+        for mask in (NotSpecified, my_asset_filter):
+            for CorrelationFactor, correlation_function in (
+                    (RollingPearsonOfReturns, pearsonr),
+                    (RollingSpearmanOfReturns, spearmanr)):
+                correlation = CorrelationFactor(
+                    baseline_asset=my_asset,
+                    returns_length=returns_length,
+                    correlation_length=correlation_length,
+                    mask=mask,
+                )
+
+                results = self.engine.run_pipeline(
+                    Pipeline(columns={'correlation': correlation}),
```

Comment:
nvm, I got mixed up with the regression case (which it looks like we're testing below).